core/src/autogluon/core/utils/feature_selection.py

```diff
@@ -0,0 +1,537 @@
+        return model, round(score, 4), round(fit_time + predict_time, 4)
+
+    def get_extra_fn_args(self, **kwargs) -> dict:
+        return {
+            'weighted': kwargs.get('weighted', True),
+            'min_fi_samples': kwargs.get('min_fi_samples', 10000),
+            'max_fi_samples': kwargs.get('max_fi_samples', 100000)
+        }
+
+    def consider_subsampling(self, X: pd.DataFrame, y: pd.Series, X_val: pd.DataFrame, y_val: pd.DataFrame, n_train_subsample: int,
+                             **kwargs) -> Tuple[pd.DataFrame, pd.Series, pd.DataFrame, pd.Series, bool, dict]:
+        """
+        If the train dataset size is larger than n_train_subsample, subsample data to make model training faster.
+        If the model is bagged and we have a lot of data, use a non-bagged version instead since it is ~10x faster to train.
+        If the validation dataset size is larger than max_fi_samples, subsample data to make model prediction faster.
+        If our task type is classification, perform stratified subsampling (rows picked proportionally to number of labels in data).
+        Update fit and predict time estimates accordingly (we only update time limit when we replace bagged model since
+        time taken within a single model fit/predict does not scale linearly with data size).
+        """
+        random_state = self.rng.integers(low=0, high=1e5)
+        max_fi_samples = kwargs.get('max_fi_samples', 100000)
+        subsample_train = n_train_subsample is not None and len(X) > n_train_subsample
+        subsample_val = max_fi_samples is not None and X_val is not None and len(X_val) > max_fi_samples
+
+        if subsample_train:
+            logger.log(30, f"\tNumber of training samples {len(X)} is greater than {n_train_subsample}. Using {n_train_subsample} samples as training data.")
+            if kwargs.get('num_classes', None) is not None:
+                # stratified sample if classification
+                combined = pd.concat([X, y], axis=1)
+                y_name = combined.columns[-1]
+                combined = combined.groupby(y_name).apply(lambda x: x.sample(frac=n_train_subsample/len(X), random_state=random_state)).reset_index(drop=True)
+                X_train, y_train = combined.drop(columns=[y_name]), combined[y_name]
+            else:
+                X_train = X.sample(n=n_train_subsample, random_state=random_state)
+                y_train = y.loc[X_train.index]
+            if kwargs.pop('replace_bag', True) and self.is_bagged and len(X) >= n_train_subsample + kwargs.get('min_fi_samples', 10000):
+                logger.log(30, f"\tThere is a lot of data and the feature selection model is bagged. Using non-bagged model.")
+                self.is_bagged = False
+                original_k_fold = kwargs['k_fold']
+                self.base_model = self.original_model.convert_to_template_child()
+                self.base_model.rename(self.original_model.name.replace('_BAG', ''))
```

Comment:
This is weird to do here, it has no relation to the other logic in this method. This is asking for bugs.